Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -228,14 +291,24 @@ void TRDGlobalTracking::run(ProcessingContext& pc)
       // skip tracks without TRD tracklets (the collision ID for the TRD tracks is initialized to -1 and only changed if a tracklet is attached to the track)
       continue;
     }
+    if (mStrict && (trdTrack.getIsAmbiguous() || trdTrack.getReducedChi2() > mTracker->Param().rec.trd.chi2StrictCut)) {
+      // skip tracks which have another hypothesis close to the best one or which do are above strict chi2 threshold
+      continue;
+    }
     nTrackletsAttached += trdTrack.getNtracklets();
     auto trackGID = trdTrack.getRefGlobalTrackId();
     if (trackGID.includesDet(GTrackID::Source::ITS)) {
       // this track is from an ITS-TPC seed
       tracksOutITSTPC.push_back(trdTrack);
+      if (mUseMC) {
+        fillMCTruthInfo(trdTrack, inputTracks.getTPCITSTrackMCLabel(trackGID), matchLabelsITSTPC, inputTracks.getTRDTrackletsMCLabels());
```

Comment:
getting `inputTracks.getTPCITSTrackMCLabel(trackGID)` in loop is more expensive (a span is created on the fly at every call) than fetching once `const auto lblSpan = inputTracks.getTPCITSTrackMCLabels()` then accessing in loop `lblSpan[trackGID]`.

Reply:
Yes, I did read the comment in RecoContainer.h and wanted to do it that way. The random access was there for testing. Above in line 202 and following I already got the span of all labels, I just forgot to change to using `itstpcTrackLabels[trackGID]`. Thanks for spotting